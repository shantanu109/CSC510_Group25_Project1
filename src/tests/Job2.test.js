import React from 'react';
import { shallow, mount } from 'enzyme';
import { Provider } from 'react-redux';
import Job2 from '../components/Job2';
import { configureStore } from '../store';
const store = configureStore(); 
const mockProps = {
  auth: {
    user: {
      email: 'test@example.com',
      _id: 'test123',
      name: 'Test User',
    },
  },
  job: [
    {
      _id: 1,
      itemname: 'Test Item 1',
      quantity: 5,
    },
  ],
  menu: {
    menuname: 'Test Menu',
  },
  order: {
    _id: 1,
    items: [
      {
        item_id: {
          menuname: 'Test Item',
        },
        quantity: 2,
      },
    ],
  },
};


test('render', () => {
    const wrapper = shallow(
      <Provider store={store}>
        <Job2 mockProps={mockProps} />
      </Provider>
    );
    expect(wrapper).toMatchSnapshot();
  });