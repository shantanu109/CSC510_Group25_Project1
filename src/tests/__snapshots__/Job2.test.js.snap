// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "@@observable": [Function],
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "subscription": Object {
        "addNestedSub": [Function],
        "getListeners": [Function],
        "handleChangeWrapper": [Function],
        "isSubscribed": [Function],
        "notifyNestedSubs": [Function],
        "onStateChange": [Function],
        "trySubscribe": [Function],
        "tryUnsubscribe": [Function],
      },
    }
  }
>
  <Connect(Job2)
    mockProps={
      Object {
        "auth": Object {
          "user": Object {
            "_id": "test123",
            "email": "test@example.com",
            "name": "Test User",
          },
        },
        "job": Array [
          Object {
            "_id": 1,
            "itemname": "Test Item 1",
            "quantity": 5,
          },
        ],
        "menu": Object {
          "menuname": "Test Menu",
        },
        "order": Object {
          "_id": 1,
          "items": Array [
            Object {
              "item_id": Object {
                "menuname": "Test Item",
              },
              "quantity": 2,
            },
          ],
        },
      }
    }
  />
</ContextProvider>
`;
